packages/jest-haste-map/src/index.ts

```diff
@@ -109,11 +111,12 @@ type Watcher = {
 
 type HasteWorker = typeof import('./worker');
 
-export type {default as FS} from './HasteFS';
-export {default as ModuleMap} from './ModuleMap';
+export const ModuleMap = HasteModuleMap as {
+  create: (rootPath: string) => IModuleMap;
+};
 export type {
-  ChangeEvent,
-  HasteMap as HasteMapObject,
```

Comment:
removing these exports is a breaking change

Reply:
specifically `HasteMapObject` probably - if people get a newer version of `jest-haste-map` than `@jest/core` it _might_ cause compilation issues.

But maybe not - since we bundle our TS types, lots of internal types are never present in `d.ts` files - https://www.runpkg.com/?@jest/core@29.0.2/build/index.d.ts

Reply:
Yeah, I *think* this is safe, because as you say it's not referenced by the built declaration files there should be no observable change to consumers of `jest`.

(It's type-breaking to users of `jest-haste-map` directly of course, but I'm working on the assumption that Jest semver is at a higher level.)